src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -178,16 +182,17 @@ private void DetermineExtensionDataProperty()
                 }
 
                 DataExtensionProperty = jsonPropertyInfo;
+                return true;
             }
+
+            return false;
         }
 
-        private JsonPropertyInfo GetPropertyThatHasAttribute(Type attributeType)
+        private JsonPropertyInfo GetPropertyThatHasAttribute(Type attributeType, Dictionary<string, JsonPropertyInfo> cache)
```

Comment:
Since this expects to find only one property, can we call this `GetUniquePropertyWithAttribute` (or something like that), or add a comment so it's clearer on first glance?

Reply:
Sure - `GetPropertyWithUniqueAttribute`